### PR TITLE
Fail messages on Data Load

### PR DESCRIPTION
**Pull request by heisenbuug**
New PR for issue  #2715

### CODE REVIEW

**Review by birm — APPROVED** [review]
This looks reasonable to me, seeing the conversation thread. Thanks for the contribution!

(Though personally, I think that a data:Load failure with the fatal flag set passing test cases probably also requires some adjustment to that function, but that's besides the point)

Does this supersede #2717?

**Comment by heisenbuug** [comment]
> This looks reasonable to me, seeing the conversation thread. Thanks for the contribution!
> 
Heyyy...
Thanks to @Abilityguy for helping me out with this...
> Does this supersede #2717?
>
Yes,

> 
> (Though personally, I think that a data:Load failure with the fatal flag set passing test cases probably also requires some adjustment to that function, but that's besides the point)
>
Oh, if we have an issue open for this, I would love to continue the work...or if you could explain a bit more, I would take that up...

File: src/mlpack/tests/feedforward_network_test.cpp [hunk 1]
```diff
@@ -160,13 +161,15 @@ TEST_CASE("FFVanillaNetworkTest", "[FeedForwardNetworkTest]")
 {
   // Load the dataset.
   arma::mat trainData;
-  data::Load("thyroid_train.csv", trainData, true);
+    if (!data::Load("thyroid_train.csv", trainData))
```
Comment (Abilityguy) [inline src/mlpack/tests/feedforward_network_test.cpp:164]:
```suggestion
  if (!data::Load("thyroid_train.csv", trainData))
```
Style fix. The indentation for the ```if``` statement is not required.

File: src/mlpack/tests/feedforward_network_test.cpp [hunk 2]
```diff
@@ -307,13 +310,15 @@ TEST_CASE("DropoutNetworkTest", "[FeedForwardNetworkTest]")
...
```
Comment (Abilityguy) [inline src/mlpack/tests/feedforward_network_test.cpp:313]:
```suggestion
  if (!data::Load("thyroid_train.csv", trainData))
```

File: src/mlpack/tests/feedforward_network_test.cpp [hunk 3]
```diff
@@ -408,13 +413,15 @@ TEST_CASE("DropConnectNetworkTest", "[FeedForwardNetworkTest]")
...
```
Comment (Abilityguy) [inline src/mlpack/tests/feedforward_network_test.cpp:416]:
```suggestion
  if (!data::Load("thyroid_train.csv", trainData))
```

File: src/mlpack/tests/feedforward_network_test.cpp [hunk 4]
```diff
@@ -497,13 +504,15 @@ TEST_CASE("FFSerializationTest", "[FeedForwardNetworkTest]")
...
```
Comment (Abilityguy) [inline src/mlpack/tests/feedforward_network_test.cpp:507]:
```suggestion
  if (!data::Load("thyroid_train.csv", trainData))
```

File: src/mlpack/tests/feedforward_network_test.cpp [hunk 5]
```diff
@@ -546,13 +555,15 @@ TEST_CASE("CustomLayerTest", "[FeedForwardNetworkTest]")
...
```
Comment (Abilityguy) [inline src/mlpack/tests/feedforward_network_test.cpp:558]:
```suggestion
  if (!data::Load("thyroid_train.csv", trainData))
```

File: src/mlpack/tests/feedforward_network_test.cpp [hunk 6]
```diff
@@ -625,13 +636,15 @@ TEST_CASE("FFNTrainReturnObjective", "[FeedForwardNetworkTest]")
...
```
Comment (Abilityguy) [inline src/mlpack/tests/feedforward_network_test.cpp:639]:
```suggestion
  if (!data::Load("thyroid_train.csv", trainData))
```

File: src/mlpack/tests/feedforward_network_test.cpp [hunk 7]
```diff
@@ -695,13 +708,15 @@ TEST_CASE("OptimizerTest", "[FeedForwardNetworkTest]")
...
```
Comment (Abilityguy) [inline src/mlpack/tests/feedforward_network_test.cpp:711]:
```suggestion
  if (!data::Load("thyroid_train.csv", trainData))
```

File: src/mlpack/tests/krann_search_test.cpp [hunk 8]
```diff
@@ -35,13 +35,16 @@ TEST_CASE("NaiveGuaranteeTest", "[KRANNTest]")
...
+  if (!data::Load("rann_test_r_3_900.csv", refData))
+    FAIL("Cannot load dataset rann_test_r_3_900.csv");
+  if (!data::Load("rann_test_q_3_100.csv", queryData))
+    FAIL("Cannot load dataset rann_test_q_3_100.csv");
 
   RASearch<> rsRann(refData, true, false, 1.0);
 
   arma::mat qrRanks;
-  data::Load("rann_test_qr_ranks.csv", qrRanks, true, false); // No transpose.
+    if (!data::Load("rann_test_qr_ranks.csv", qrRanks, false, false)) // No transpose.
```
Comment (Abilityguy) [inline src/mlpack/tests/krann_search_test.cpp:46]:
```suggestion
  if (!data::Load("rann_test_qr_ranks.csv", qrRanks, false, false)) // No transpose.
```

File: src/mlpack/tests/feedforward_network_test.cpp [hunk 9]
```diff
@@ -497,13 +504,15 @@ TEST_CASE("FFSerializationTest", "[FeedForwardNetworkTest]")
...
+      FAIL("Cannot open thyroid_train.csv");
```
Comment (Abilityguy) [inline src/mlpack/tests/feedforward_network_test.cpp:508]:
```suggestion
    FAIL("Cannot open thyroid_train.csv");
```

File: src/mlpack/tests/feedforward_network_test.cpp [hunk 10]
```diff
@@ -546,13 +555,15 @@ TEST_CASE("CustomLayerTest", "[FeedForwardNetworkTest]")
...
```
Comment (Abilityguy) [inline src/mlpack/tests/feedforward_network_test.cpp:559]:
```suggestion
    FAIL("Cannot open thyroid_train.csv");
```

File: src/mlpack/tests/feedforward_network_test.cpp [hunk 11]
```diff
@@ -625,13 +636,15 @@ TEST_CASE("FFNTrainReturnObjective", "[FeedForwardNetworkTest]")
...
```
Comment (Abilityguy) [inline src/mlpack/tests/feedforward_network_test.cpp:640]:
```suggestion
    FAIL("Cannot open thyroid_train.csv");
```

**Review by Abilityguy — CHANGES_REQUESTED** [review]
Hi @heisenbuug, 
Looks good to me except for a few style fixes which I have pointed out.

File: src/mlpack/tests/feedforward_network_2_test.cpp [hunk 12]
```diff
@@ -65,7 +65,8 @@ TEST_CASE("RBFNetworkTest", "[FeedForwardNetworkTest]")
 {
   // Load the dataset.
   arma::mat trainData;
-  data::Load("thyroid_train.csv", trainData, true);
+  if (!data::Load("thyroid_train.csv", trainData))
+    Fail("Cannot open thyroid_train.csv");
```
Comment (zoq) [inline src/mlpack/tests/feedforward_network_2_test.cpp:69]:
```suggestion
    FAIL("Cannot open thyroid_train.csv");
```

File: src/mlpack/tests/feedforward_network_2_test.cpp [hunk 13]
```diff
@@ -77,7 +78,8 @@ TEST_CASE("RBFNetworkTest", "[FeedForwardNetworkTest]")
   }
 
   arma::mat testData;
-  data::Load("thyroid_test.csv", testData, true);
+  if (!data::Load("thyroid_test.csv", testData))
+    Fail("Cannot open thyroid_test.csv");
```
Comment (zoq) [inline src/mlpack/tests/feedforward_network_2_test.cpp:82]:
```suggestion
    FAIL("Cannot open thyroid_test.csv");
```

**Review by zoq — COMMENTED** [review]
Looks good to me as well, just some minor comments.

File: src/mlpack/tests/feedforward_network_2_test.cpp (hunk 12 above)
Comment (rcurtin) [reply src/mlpack/tests/feedforward_network_2_test.cpp:69]: I thought it was strange that the tests were still passing, but then I noticed this file wasn't being compiled.  I opened #2851, which we should merge and then make sure this still works before merging this one. :+1:

**Review by rcurtin — CHANGES_REQUESTED** [review]
Looks good to me!  Thank you @heisenbuug!  I requested changes just so that we can ensure that #2851 gets merged and then we can re-run the tests here, just to ensure that nothing is wrong with `feedforward_network_2_test.cpp`. :+1:

**Comment by zoq** [comment]
@mlpack-jenkins test this please

**Review by jeffin143 — APPROVED** [review]

**Comment by zoq** [comment]
Alright, we can ignore the static analysis issue, it's a false warning.

**Review by zoq — APPROVED** [review]
Thanks for putting this together :+1:

**Review by shrit — APPROVED** [review]

**Comment by zoq** [comment]
> Looks good to me! Thank you @heisenbuug! I requested changes just so that we can ensure that #2851 gets merged and then we can re-run the tests here, just to ensure that nothing is wrong with `feedforward_network_2_test.cpp`.

Since, #2851 is merged and we re-run the tests here and everything is fine, let's merge this one as well.

**Review by rcurtin — APPROVED** [review]
Awesome, thank you for these changes!

**Pull request merged**